interfaces/apparmor/backend.go

```diff
@@ -103,7 +103,26 @@ func (b *Backend) Setup(snapInfo *snap.Info, developerMode bool, repo *interface
 	if errReload != nil {
 		return errReload
 	}
-	return errUnload
+	if errUnload != nil {
+		return errUnload
+	}
+	// Sanity checking, see if all the profiles are really loaded and load
+	// those profiles that are not. This can happen if the files were on disk
+	// but were not really loaded (e.g. due to an interrupted install).
```

Comment:
@jdstrand this is exactly what this patch is doing. The tricky part is part is that apparmor backend code only loads / unloads profiles in reaction changes to profiles on disk.

Thinking about it now I think I will unconditionally load all profiles. This will take care of apparmor include file changes (``apparmor_parser -p`` discussion from the email thread) and will be a simple thing to do before we invest in a more efficient version.